robot/impl/resource_manager.go

```diff
@@ -185,29 +185,22 @@ func (manager *resourceManager) updateRemoteResourceNames(
 	manager.logger.CDebugw(ctx, "updating remote resource names", "remote", remoteName)
 	activeResourceNames := map[resource.Name]bool{}
 	newResources := rr.ResourceNames()
-	oldResources := manager.remoteResourceNames(remoteName)
-	for _, res := range oldResources {
+	if newResources == nil {
+		// nil implies our connection to the remote is currently broken. Return without changing any
+		// state for this remote.
+		return false
+	}
+
+	// Initialize a map with all existing resources as a map item. We will iterate through all of
+	// the `newResources` and set the map value for that item to true. Anything left with `false` at
+	// the end has been removed and will be marked for removal.
+	for _, res := range manager.remoteResourceNames(remoteName) {
 		activeResourceNames[res] = false
 	}
 
 	anythingChanged := false
-
 	for _, resName := range newResources {
 		remoteResName := resName
-		res, err := rr.ResourceByName(remoteResName) // this returns a remote known OR foreign resource client
```

Comment:
Nothing changed here. Just code movement. What's important is this is now below [oldline 215/newline 208](https://github.com/viamrobotics/rdk/pull/4268/files#diff-77fd7ff51c1b7804c67c4df0a9a672ea0e3842b4a54bbd8ef81af82ac95c4f22L215). I hope the comment I added to the new code outlines the significance.